// streamlink-serverless/lib/index.js
"use strict";
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Streamlink = void 0;
const JSII_RTTI_SYMBOL_1 = Symbol.for("jsii.rtti");
const path_1 = require("path");
const cdk = require("aws-cdk-lib");
const aws_cdk_lib_1 = require("aws-cdk-lib");
const lambda = require("aws-cdk-lib/aws-lambda");
const aws_lambda_1 = require("aws-cdk-lib/aws-lambda");
const constructs_1 = require("constructs");
class Streamlink extends constructs_1.Construct {
    constructor(scope, id, _props = {}) {
        super(scope, id);
        const runtime = aws_lambda_1.Runtime.PYTHON_3_9;
        const streamlinkApp = path_1.join(__dirname, '..', 'streamlink');
        const layer = new lambda.LayerVersion(this, 'Layer', {
            compatibleRuntimes: [runtime],
            code: lambda.Code.fromAsset(streamlinkApp, {
                bundling: {
                    image: aws_cdk_lib_1.DockerImage.fromBuild(streamlinkApp),
                    workingDirectory: '/build',
                    command: [
                        'bash', '-c',
                        `pip install --upgrade aws_lambda_types streamlink lxml -t python/lib/${runtime.name}/site-packages/ && zip -rq /asset-output/layer.zip python`,
                    ],
                    outputType: aws_cdk_lib_1.BundlingOutput.ARCHIVED,
                },
            }),
        });
        this.function = new lambda.Function(this, 'Default', {
            runtime,
            handler: 'app.handler',
            code: lambda.Code.fromAsset(streamlinkApp),
            memorySize: 512,
            timeout: cdk.Duration.seconds(5),
            layers: [layer],
        });
    }
}
exports.Streamlink = Streamlink;
_a = JSII_RTTI_SYMBOL_1;
Streamlink[_a] = { fqn: "streamlink-serverless.Streamlink", version: "0.0.70" };
